be/src/exec/pipeline/source_operator.h

```diff
@@ -39,11 +39,11 @@ class SourceOperatorFactory : public OperatorFactory {
     // OlapScanOperator needs to do so.
     virtual bool with_morsels() const { return false; }
     // Set the DOP(degree of parallelism) of the SourceOperator, SourceOperator's DOP determine the Pipeline's DOP.
-    virtual void set_degree_of_parallelism(size_t degree_of_parallelism) {
-        this->_degree_of_parallelism = degree_of_parallelism;
-    }
+    void set_degree_of_parallelism(size_t degree_of_parallelism) { _degree_of_parallelism = degree_of_parallelism; }
```

Comment:
limit its range?[0,xx?]

Reply:
This is just a set method. IMO, the simple set method shouldn't have the other work beside setting field.
And `SourceOperator` doesn't know the DOP range, the DOP range should be considered by the caller, not the callee.